module3/projects/consultancy/index.md

```diff
@@ -0,0 +1,88 @@
+
+## Phases
+
+* [Ideation](./ideation)
+* [Inception](./inception)
+* [Development](./development)
+* [Retro](./retro_guide)
+* [Presentation](./presentation)
+* [Evaluation](./evaluation)
+
+## Technical Requirements
+
+You have a lot of freedom with this project, but there are a few technical requirements. Overall, your project must:
+
+1. Includes at least one Rails API and one ReactJS application.
+2. Use one external API, but it can be consumed on the FE or BE. 
+3. Be deployed
+4. Assume one single user - multiple users with log in and authentication is out of scope.
+5. Implement functionality beyond simply data display - there needs to be some way for the user to work with or manipulate the data (favoriting, searching, filtering, commenting, rating, etc)
+
+<section class="dropdown">
+### Back End Requirements
+
+* Thorough testing at both the unit and integration level.
+  * Test Coverage tool like SimpleCov is used
+  * Happy Path and Sad Path/Edge Case testing is included
+* One Stretch Tech option (listed below) is implemented
+  * **Caching**: Using low-level caching (no FE caching). Team should be able to identify how the points where caching is implemented help to alleviate the main workflow of the application, and ideally show before/after data. 
+  * **Background Jobs**: Consider using Sidekiq with Redis. Team should be able to identify how the background worker(s) alleviate the main workflow of the application, and ideally show before/after data. 
+  * **CirlceCI**: Team should be able to identify the use case of this technology, its pros and cons. In order to receive credit for this stretch tech, teams must not only set it up, but incorporate continuous integration principles into their workflow (i.e. when the build is red, discussion about fixing the build is visible, and commits are not pushed on top of a red build)
+  * **Rate Limiting**: Consider using the gem called "rack-attack" and test to demonstrate it's working! Team should be able to discuss why rate limiting is an effective feature. Ideally, be able to demonstrate it live using a small script (either live or in production). 
+  * **Microservice**: Team should create a separate microservice (separate from their BE API app) using Sinatra or a different framework, and be able to speak to the pros/cons of that framework when compared to Rails. 
+
+Other new tools/concepts that would be fun but not considered your stretch-tech (think post-project exploration):
+* Pagination
+* [Security Vulnerability Scanner](https://github.com/presidentbeef/brakeman)
+* OpenAPI Documentation
+* [Circuit Breaker Design Pattern](https://makisushi.io/posts/asd)
+* Observability (i.e. New Relic, Skylight, Honey Badger, Rails Performance)
+
+</section>
+
+<section class="dropdown">
+### Front End
+
+* Utilizes React Router
+* Includes at least one POST request to the BE
+* Robust Cypress testing
+<!-- TO-DO: Require intercepts with Cypress tests? -->
```

Comment:
Yes, Cypress testing is not complete without intercepts.

Reply:
In previous projects like this one ([Stretch Tech](https://frontend.turing.edu/projects/module-3/stretch.html)), we've made it really clear what amount of interactivity is necessary to be considered a solid MVP. Like, they can't just show us data, but they need to actually be able to manipulate it. I think that having POST called out helps with that, but we could be more detailed.

Reply:
```
* The application should be a multi-page application using React Router
* The application must include at least one POST to the backend
* The application cannot simply be a display of data - there needs to be some way for the user to work with or manipulate the data (favoriting, searching, filtering, commenting, rating, etc)
* The must be robust Cypress testing, with all network requests properly intercepted and all user flows tested
```

Reply:
Also, question: We usually include "Assume one single user - multiple users with log in and authentication is out of the scope of this project." for their Stretch Tech project. Do we want to include that here, too?

Reply:
Yes! The "assume one single user" is listed in the main technical requirements page! I stole the language straight from the stretch tech spec